src/pubtools/_marketplacesvm/tasks/delete/command.py

```diff
@@ -0,0 +1,267 @@
+
+        Yields:
+            The AmiPushItems from the given sources.
+        """
+        for source_url in self.args.source:
+            with Source.get(source_url) as source:
+                log.info("Loading items from %s", source_url)
+                for item in source:
+                    if not isinstance(item, AmiPushItem):
+                        log.warning(
+                            "Push Item %s at %s is not an AmiPushItem, dropping it from the queue.",
+                            item.name,
+                            item.src,
+                        )
+                        continue
+                    self.builds_borg.received_builds.add(item.build_info.id)
+                    yield item
+
+    @property
+    def mapped_items(self) -> List[AmiPushItem]:  # type: ignore [override]
+        """
+        Return the mapped push item with destinations and metadata from StArMap.
+
+        Returns
+            The wrapped push item with the additional information from StArMap.
+        """
+        mapped_items = []
+        for item in self.raw_items:
+            mapped_items.append(item)
+        return mapped_items
+
+    def update_rhsm_metadata(self, push_item: AmiPushItem) -> None:
+        """
+        Update image in RHSM to 'invisible'.
+
+        Args:
+            push_item (AmiPushItem)
+                Push item to pull information from.
+            provider (str)
+                Provider that the image resides at.
+        """
+        if self.rhsm_image_ids and push_item.image_id not in self.rhsm_image_ids:
+            log.warning("AMI image: %s not found, skipping update in rhsm.", push_item.image_id)
+        else:
+            try:
+                img_type = push_item.type or ""
+                product = self.get_rhsm_product(
+                    push_item.release.product, img_type, push_item.marketplace_entity_type
+                )
+            except RuntimeError:
+                log.info("%s not found in RHSM", push_item.release.product)
+                return
+            image_meta = {
+                "image_id": push_item.image_id,
+                "image_name": push_item.name,
+                "arch": push_item.release.arch,
+                "product_name": product["name"],
+                "version": push_item.release.version or None,
+                "variant": push_item.release.variant or None,
+                "status": "invisible",
+            }
+            if self.args.dry_run:
+                log.info("Would have updated image %s in rhsm", push_item.image_id)
+                return
+
+            log.info("Attempting to update the existing image %s in rhsm", push_item.image_id)
+            out = self.rhsm_client.aws_update_image(**image_meta)
+            resp = out.result()
+            if not resp.ok:
+                log.error("Failed updating image %s", push_item.image_id)
+
+            log.info("Existing image %s succesfully updated in rhsm", push_item.image_id)
```

Comment:
I think this will print both "Failed" and "Success" messages in the logs

You can either use and `else` or raise:

Example 1:

```
            if not resp.ok:
                log.error("Failed updating image %s", push_item.image_id)
            else:
                log.info("Existing image %s succesfully updated in rhsm", push_item.image_id)

```

Example 2:
```
            if not resp.ok:
                err_msg = "Failed updating image %s", push_item.image_id
                log.error(err_msg)
                raise RuntimeError(err_msg)

            log.info("Existing image %s succesfully updated in rhsm", push_item.image_id)

```

Reply:
whoops meant to just return

Reply:
tyvm